tests/helper/helper_oc.go

```diff
@@ -450,10 +450,25 @@ func (oc *OcRunner) WaitForDCRollout(dcName string, project string, timeout time
 	session.Wait(timeout)
 }
 
-// CheckForExistence checks if the given resource type exists on the cluster
-func (oc *OcRunner) CheckForExistence(resourceType, namespace string) {
-	session := CmdRunner(oc.path, "get", resourceType, "--namespace", namespace)
-	Eventually(session).Should(gexec.Exit(0))
-	output := string(session.Wait().Out.Contents())
-	Expect(strings.ToLower(output)).To(ContainSubstring(""))
+// WaitAndCheckForExistence wait for the given and checks if the given resource type gets deleted on the cluster
+func (oc *OcRunner) WaitAndCheckForExistence(resourceType, namespace string, timeoutMinutes int) bool {
+	pingTimeout := time.After(time.Duration(timeoutMinutes) * time.Minute)
+	// this is a test package so time.Tick() is acceptable
+	// nolint
+	tick := time.Tick(time.Second)
+	for {
+		select {
+		case <-pingTimeout:
+			Fail(fmt.Sprintf("Timeout out after %v minutes", timeoutMinutes))
+
+		case <-tick:
+			session := CmdRunner(oc.path, "get", resourceType, "--namespace", namespace)
+			Eventually(session).Should(gexec.Exit(0))
+			output := string(session.Wait().Err.Contents())
```

Comment:
@mik-dass This looks weird as the ```oc``` command returns exit code ```0``` and then you are checking ```Err.Contents()``` for validation message. AFAIK if command returns exit code ```0``` then the output channels to ```stdout```, however here in case of oc it does not happen. I would suggest create an upstream issue against oc and mark the issue here in the comment plus some few context why you need the workaround of ```output := string(session.Wait().Err.Contents())``` atm.

Reply:
Added https://github.com/openshift/odo/pull/2750/files#diff-5397c9b14c0556bf39518ae854392472R467

Reply:
/lgtm